apps/frontend/src/app/5_pages/PortfolioPage/components/AssetSection/components/AssetBalanceRow/AssetBalanceRow.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { FC, useEffect } from 'react';
+
+import { SupportedTokens } from '@sovryn/contracts';
+import { Paragraph } from '@sovryn/ui';
+
+import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { AssetRenderer } from '../../../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { getTokenLongName } from '../../../../../../../constants/tokens';
+import { useAssetBalance } from '../../../../../../../hooks/useAssetBalance';
+import { useDollarValue } from '../../../../../../../hooks/useDollarValue';
+import { getRskChainId } from '../../../../../../../utils/chain';
+import { getCurrencyPrecision } from '../../AssetSection.utils';
+import styles from './AssetBalanceRow.module.css';
+
+type AssetBalanceRowProps = {
+  token: SupportedTokens;
+  updateUSDValue: (usdValue: string) => void;
+};
+
+export const AssetBalanceRow: FC<AssetBalanceRowProps> = ({
+  token,
+  updateUSDValue,
+}) => {
+  const { weiBalance, balance } = useAssetBalance(token, getRskChainId());
+  const { usdValue } = useDollarValue(token, weiBalance);
+
+  useEffect(() => {
+    updateUSDValue(usdValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
just a note, Ideally I would try to refactor here and do it without disabling dependencies, but I don't think that's critical at the moment